### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by triblondon**
The contributor guide is getting quite long, and people were starting to find it confusing. With an increasing number of contributors, I think it's a good time to revise the contributor guide, and split it into sections.  Proofing of the new text much appreciated.  Does it make sense? Are there gaps? Gotchas? Is it easy to follow?

The best thing to do here is to checkout this branch and view the new docs through the service, rather than trying to read the diff, as I've moved stuff around.  To make this easier I've deployed it to the QA server: http://qa.polyfill.io.

New homepage:

image

### CODE REVIEW

File: docs/contributing/authoring-polyfills.html [hunk 1]
```diff
@@ -0,0 +1,173 @@
+{{>header}}
+
+{{>nav this}}
+
+	<div class="o-techdocs-main o-techdocs-content">
+
+		<h1>Authoring polyfills</h1>
+
+		<p>Polyfills are located in the <code>/polyfills</code> directory and organised by feature name.</p>
```
Comment (jonathan-fielding) [inline docs/contributing/authoring-polyfills.html:9]: Would be useful to know what to call the polyfill, example being that when I did the picture polyfill pull request you asked me to move it to HTMLPictureElement, so something that suggests the folder/polyfill name might be useful

File: docs/contributing/authoring-polyfills.html [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+		<h2 id='files'>Files</h2>
+
+		<p>Each polyfill directory (and subdirectory) is structured like this:</p>
+
+		<ul>
+			<li><code>polyfill.js</code>: Required. Code to apply the polyfill behaviour.  See <a href="#configuration">configuration</a>.</li>
+			<li><code>detect.js</code>: A single expression that returns true if the feature is present in the browser (and the polyfill is therefore not required), false otherwise.  Can be an IIFE.  If not present, polyfill cannot be gated (wrapped in a feature-detect).  <strong>Must not have a trailing semicolon</strong>.</li>
+			<li><code>tests.js</code>: A set of tests written using <a href="http://mochajs.org/">mocha</a> and <a href="https://github.com/LearnBoost/expect.js/">expect</a>, to test the feature.  See <a href="#test-authoring-guidelines">test authoring guidelines</a>.</li>
+			<li><code>update.task.js</code>: Optional alternative to <code>polyfill.js</code>. This file is used in combination with the Grunt task <code>updatelibrary</code> to create a local copy of an external polyfill.</li>
```
Comment (jonathan-fielding) [inline docs/contributing/authoring-polyfills.html:27]: Should this now be mandatory for all third party polyfills?

File: docs/contributing/authoring-polyfills.html [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+}</code></pre>
+
+		<p>A request from IE7 would receive this polyfill, since it is targeted to IE 6-9.  It <em>may</em> also receive polyfills for <code>Object.defineProperties</code> and <code>Object.create</code>, since those are dependencies of the polyfill in this example, if those polyfills also apply in IE7.</p>
+
+		<h2 id="authoring-guidelines">Authoring guidelines</h2>
+
+		<p>We are always glad to accept contributions of original polyfills as well as copies of third party polyfills maintained elsewhere.  It's important to distinguish between these, because:</p>
+
+		<ul>
+		<li><strong>original polyfills</strong> are those where the polyfill service is the canonical source of the code.  They do not have a <code>repo</code> or <code>license</code> property in their config.  Original polyfills:<ul>
```
Comment (jonathan-fielding) [inline docs/contributing/authoring-polyfills.html]: fix indenting of the li

File: docs/contributing/authoring-polyfills.html [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+			<li>may contain a feature detect.  Ideally they wouldn't, because the service will wrap the polyfill in a detect, but we'll tolerate a detect in order to maintain sync with the upstream source, provided that it is strict enough to fail in the case of broken implementations, as well as missing ones.</li>
+		</ul></li>
+		</ul>
+
+		In addition, whether original or third party, if the polyfill targeting is open ended (ie it targets current versions of browsers), it <em>must</em> only make changes that create conformance with a published specification.  If it is targeted <em>only</em> at older versions of browsers, it <em>may</em> be OK to exhibit any behaviour that is likely to satisfy the needs of applications that depend on the feature.  For example, our <code>devicePixelRatio</code> polyfill in very old browsers simply returns <code>1</code> without actually measuring anything.
```
Comment (jonathan-fielding) [inline docs/contributing/authoring-polyfills.html]: missing 'p' tag arround the text

File: docs/contributing/authoring-polyfills.html [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+};</code></pre>
+
+		<p>Often changes are required (or desirable) to support the format of polyfills needed by the polyfill service, for example:</p>
+
+		<ul>
+		<li><strong>Private dependencies:</strong> some polyfill authors will create polyfill collections that abstract common tasks into shared functions that are not themselves public web platform APIs.  The polyfill service supports this via private polyfills (which are prefixed <code>_</code>), and will import and declare them in the parent scope so that the polyfill can consider them to be globals.  Polyfills may therefore need to be amended to remove import or require statements (or to provide an import or require implementation that will satisfy them), and to provide the necessary dependency via other means (either as a seaparate private polyfill or by inlining it)</li>
+		<li><strong>Embedded polyfills:</strong> some polyfills include code that fills more than one feature. For example, <code>classList</code> polyfills typically also polyfill <code>DOMTokenList</code>.  This should be a dependency relationship expressed using the <code>config.json</code> metadata, rather than conflating two polyfills into one.</li>
```
Comment (jonathan-fielding) [inline docs/contributing/authoring-polyfills.html]: indentation of the li within the ul

File: docs/contributing/authoring-polyfills.html [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+			<tr><td><code>safari</code></td><td>Safari</td><td>{{baselines.safari}}</td></tr>
+			<tr><td><code>ios_saf</code></td><td>Safari on iOS</td><td>{{baselines.ios_saf}}</td></tr>
+			<tr><td><code>firefox</code></td><td>Firefox</td><td>{{baselines.firefox}}</td></tr>
+			<tr><td><code>firefox_mob</code></td><td>Firefox on Android</td><td>{{baselines.firefox_mob}}</td></tr>
+			<tr><td><code>android</code></td><td>Android Browser</td><td>{{baselines.android}}</td></tr>
+			<tr><td><code>opera</code></td><td>Opera</td><td>{{baselines.opera}}</td></tr>
+			<tr><td><code>op_mob</code></td><td>Opera Mobile</td><td>{{baselines.op_mob}}</td></tr>
+			<tr><td><code>op_mini</code></td><td>Opera Mini</td><td>{{baselines.op_mini}}</td></tr>
+			<tr><td><code>bb</code></td><td>Blackberry</td><td>{{baselines.bb}}</td></tr>
+			</tbody>
```
Comment (jonathan-fielding) [inline docs/contributing/authoring-polyfills.html:163]: is it the code style you are using to not indent inside the tbody and thead

File: docs/contributing/common-scenarios.html [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+			<li>State your agreement to our contribution terms.</li>
+		</ol>
+
+
+		<h3 id='new-browser'>The service is incorrectly identifying a browser</h3>
+
+		<p>Although there are only five significant rendering engines (blink, presto, trident, gecko and webkit), there are lots of distinct browsers and user agents. We don't currently support anywhere near all of them, but are happy to add support on demand.</p>
+
+		<ol>
+			<li>Find out the <code>User-Agent</code> header value sent by the browser</li>
```
Comment (jonathan-fielding) [inline docs/contributing/common-scenarios.html]: include a site that will tell you your user agent such as http://www.whatsmyua.com/

File: docs/contributing/common-scenarios.html [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+{{>nav this}}
+
+	<div class="o-techdocs-main o-techdocs-content">
+
+		<h1>Common contribution scenarios</h1>
+
+		<p>A few scenarios come up time and again on our issue tracker. If you want to contribute, and one of the scenarios below descirbes what you want to change, follow the steps to acheive a painless and rapidly merged pull request!</p>
+
+
+		<h3 id='adding-browsers'>A particular polyfill is being served to the wrong browsers / not enough browsers</h3>
```
Comment (jonathan-fielding) [inline docs/contributing/common-scenarios.html:12]: why are we going straight from a h1 to a h3?

**Comment by jonathan-fielding** [comment]
General comments - I ran the pages through the w3c spell checker and nothing was flagged, one thing I did notice though is that the use of 'eg' is written as 'eg' which sometimes makes it harder to read, consider using e.g. 

On the whole this pull request is a big improvement and does make the reading of the contribution guidelines a lot clearer :smile:

File: docs/contributing/authoring-polyfills.html [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+		<h2 id="browsers">Browsers</h2>
+
+		<p>The short names should be used in the <code>config.json</code> to configure the browser support using the <code>browsers</code> key.</p>
+
+		<table>
+			<thead>
+			<tr><th>Short name</th><th>User Agent Name</th><th>Baseline</th></tr>
+			</thead>
+			<tbody>
+			<tr><td><code>ie</code></td><td>Internet Explorer / Edge / Edge Mobile</td><td>{{baselines.ie}}</td></tr>
```
Comment (jonathan-fielding) [inline docs/contributing/authoring-polyfills.html:150]: The baseline data isn't showing on on the QA site, not sure if this is environment issue or an issue caused by this pull request

File: docs/contributing/authoring-polyfills.html [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+			<li><code>update.task.js</code>: Optional alternative to <code>polyfill.js</code>. This file is used in combination with the Grunt task <code>updatelibrary</code> to create a local copy of an external polyfill.</li>
+			<li><code>config.json</code>: A config file conforming to the spec below</li>
+		</ul>
+
+		<h2 id='configuration'>Configuration</h2>
+
+		<p>The config.json file may contain any of the following keys:</p>
+
+		<ul>
+			<li><code>browsers</code>: Object, one key per browser family name (see <a href="#browsers">browser support</a>), with the value forming either a range or a list of specific versions separated by double pipes, idenitfying the versions to which the polyfill <em>should be applied</em>. We support all valid <a href="https://github.com/npm/node-semver#ranges">node-semver ranges</a>, but discourage the use of the `<=` operator as it tends to cause confusion.</li>
```
Comment (jonathan-fielding) [inline docs/contributing/authoring-polyfills.html]: typo: 'idenitfying' should be 'identifying'

File: docs/contributing/authoring-polyfills.html [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+	// ... download/import latest source here
+	// ... do any required source modification
+	var newSource = 'test';
+	grunt.file.write(path.join(__dirname, 'polyfill.js'), newSource);
+};</code></pre>
+
+		<p>Often changes are required (or desirable) to support the format of polyfills needed by the polyfill service, for example:</p>
+
+		<ul>
+		<li><strong>Private dependencies:</strong> some polyfill authors will create polyfill collections that abstract common tasks into shared functions that are not themselves public web platform APIs.  The polyfill service supports this via private polyfills (which are prefixed <code>_</code>), and will import and declare them in the parent scope so that the polyfill can consider them to be globals.  Polyfills may therefore need to be amended to remove import or require statements (or to provide an import or require implementation that will satisfy them), and to provide the necessary dependency via other means (either as a seaparate private polyfill or by inlining it)</li>
```
Comment (jonathan-fielding) [inline docs/contributing/authoring-polyfills.html]: typo: seaparate should be separate

File: docs/contributing/common-scenarios.html [hunk 12]
```diff
@@ -0,0 +1,94 @@
+{{>header}}
+
+{{>nav this}}
+
+	<div class="o-techdocs-main o-techdocs-content">
+
+		<h1>Common contribution scenarios</h1>
+
+		<p>A few scenarios come up time and again on our issue tracker. If you want to contribute, and one of the scenarios below descirbes what you want to change, follow the steps to acheive a painless and rapidly merged pull request!</p>
```
Comment (jonathan-fielding) [inline docs/contributing/common-scenarios.html]: typos 

descirbes should be describes
acheive should be achieve

File: docs/contributing/common-scenarios.html [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+
+		<ol>
+			<li>Find out the <code>User-Agent</code> header value sent by the browser</li>
+			<li>Load <code>https://cdn.polyfill.io/v2/polyfill.js?ua={UAHEADER}</code> and take note of the detected, normalised browser name (eg 'chrome/40') that is displayed in the comment at the top of the output</li>
+			<li>Work out which of our existing families the browser matches, eg Firefox for iOS is most usefully seen as an alias of Safari for iOS.</li>
+			<li>Make the apppropriate change:
+			<ul>
+				<li>If the browser is misrecognised as another browser that we support (eg niche Acme browser is recognised as 'chrome' but it's actually gecko based), the family is right but the version number is wrong, or it's coming up as 'unknown', this is probably a problem with the node useragent module. Check our <code>UA.js</code> module to ensure we don't have any craxy aliasing rules that are misbehaving, then raise your issue in the useragent repo, not here.</li>
```
Comment (jonathan-fielding) [inline docs/contributing/common-scenarios.html]: typo: craxy should be crazy

**Comment by jonathantneal** [comment]
I could have forgotten something, but why no mention of CC0 and the recommendation of MIT?

File: docs/contributing/authoring-polyfills.html (hunk 2 above)
Comment (triblondon) [inline docs/contributing/authoring-polyfills.html:27]: Lots of mandatory requirements reduces demand to participate, so we have to strike a balance.

File: docs/contributing/authoring-polyfills.html (hunk 6 above)
Comment (triblondon) [inline docs/contributing/authoring-polyfills.html:163]: Not especially. I'm pretty relaxed about indent conventions for markup.

File: docs/contributing/common-scenarios.html (hunk 8 above)
Comment (triblondon) [inline docs/contributing/common-scenarios.html:12]: The headings are long, and I didn't want them to appear in the left nav.  Fairly lousy reason but it does produce a better UX for most.

**Comment by triblondon** [comment]
Thanks @jonathan-fielding, that's great.  @jonathantneal I would prefer outside contributors to use MIT because:
- It's shorter than CC0 - around 4x fewer words in the licence
- Most polyfills we absorb into the service are MIT licensed already, so it is more consistent
- CC0 actually confers more restrictions on use than MIT - for example if the work contains any trademarks, those are excluded. MIT gives confidence that the entire work is covered by the same rules.  See [CC0](http://choosealicense.com/licenses/cc0-1.0/) vs [MIT](http://choosealicense.com/licenses/mit/) in detail.
- The only requirement that MIT has which CC0 does not is the requirement to state the license. We should do this anyway as a matter of course because otherwise you wouldn't know what the copyright status of the work was.

@jonathantneal are you able to review the rest of the content?

File: docs/contributing/authoring-polyfills.html (hunk 6 above)
Comment (jonathan-fielding) [inline docs/contributing/authoring-polyfills.html:163]: Fair enough, might make it easier to read however if it were indented, :D

File: docs/contributing/authoring-polyfills.html [hunk 14]
```diff
@@ -0,0 +1,181 @@
...
+
+		<p>If a feature has no JavaScript API, e.g. support for CSS styling of HTML5 elements, it should be given a descriptive name prefixed with a tilde (~), ie. <code>~html5-elements</code>.</p>
+
+		<h2 id='files'>Files</h2>
+
+		<p>Each polyfill directory (and subdirectory) is structured like this:</p>
+
+		<ul>
+			<li><code>polyfill.js</code>: Required. Code to apply the polyfill behaviour.  See <a href="#configuration">configuration</a>.</li>
+			<li><code>detect.js</code>: A single expression that returns true if the feature is present in the browser (and the polyfill is therefore not required), false otherwise.  Can be an IIFE.  If not present, polyfill cannot be gated (wrapped in a feature-detect).  <strong>Must not have a trailing semicolon</strong>.</li>
```
Comment (j-mes) [inline docs/contributing/authoring-polyfills.html]: Is IIFE intentionally missing a link explaining what the abbreviation or what it does?

Would I be correct in assuming that it is based on what [Wikipedia](https://en.wikipedia.org/wiki/Immediately-invoked_function_expression) has to say about IIFE.

Comment (triblondon) [inline docs/contributing/authoring-polyfills.html]: Yes!  Thanks for pointing it out. Looks like Jake jumped in like a champ to fix it.

File: docs/contributing/authoring-polyfills.html [hunk 15]
```diff
@@ -0,0 +1,181 @@
...
+				<li>may contain a feature detect.  Ideally they wouldn't, because the service will wrap the polyfill in a detect, but we'll tolerate a detect in order to maintain sync with the upstream source, provided that it is strict enough to fail in the case of broken implementations, as well as missing ones.</li>
+			</ul></li>
+		</ul>
+
+		<p>In addition, whether original or third party, if the polyfill targeting is open ended (ie it targets current versions of browsers), it <em>must</em> only make changes that create conformance with a published specification.  If it is targeted <em>only</em> at older versions of browsers, it <em>may</em> be OK to exhibit any behaviour that is likely to satisfy the needs of applications that depend on the feature.  For example, our <code>devicePixelRatio</code> polyfill in very old browsers simply returns <code>1</code> without actually measuring anything.</p>
+
+		<h3 id="binding">Binding to global scope</h3>
+
+		<p>Polyfill bundles produced by the service are wrapped in the following IIFE:</p>
```
Comment (j-mes) [inline docs/contributing/authoring-polyfills.html]: This could do with another abbr tag for IIFE.

**Comment by j-mes** [comment]
Having read through the docs twice. I would say as I'm not really familiar with the service or how to use the polyfills service, the documentation is a very good starting point. I especially like the live examples because then someone could get the ball running, and refer back to the documentation if something doesn't work to the expected outcome.

**Comment by triblondon** [comment]
Thanks James.  I think that's a decent enough review now, let's do it.

**Pull request merged**